### ISSUE TITLE
armhf: Debian 10: ERROR: Failed building wheel for cmake 

### ISSUE THREAD

**Issue by leamas** [scikit-build/cmake-python-distributions#192]
As heading says: On Debian 10/buster trying to install cmake using pypi I run into problems:

```
Building wheel for cmake (PEP 517): started
  Building wheel for cmake (PEP 517): finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 /root/.local/lib/python3.7/site-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /tmp/tmp027pwjbz

       cwd: /tmp/pip-install-_r4xwown/cmake_82599890a8a04506a314588eb24cfeef

  Complete output (9 lines):
    File "/tmp/pip-build-env-jqdgcx1z/overlay/lib/python3.7/site-packages/skbuild/setuptools_wrap.py", line 564, in setup
      cmkr = cmaker.CMaker(cmake_executable)
    File "/tmp/pip-build-env-jqdgcx1z/overlay/lib/python3.7/site-packages/skbuild/cmaker.py", line 95, in __init__
      self.cmake_version = get_cmake_version(self.cmake_executable)
    File "/tmp/pip-build-env-jqdgcx1z/overlay/lib/python3.7/site-packages/skbuild/cmaker.py", line 82, in get_cmake_version
      "Problem with the CMake installation, aborting build. CMake executable is %s" % cmake_executable)
  Traceback (most recent call last):

  Problem with the CMake installation, aborting build. CMake executable is cmake
  ----------------------------------------
  ERROR: Failed building wheel for cmake

Failed to build cmake

ERROR: Could not build wheels for cmake which use PEP 517 and cannot be installed directly

```

This is part of a larger build. Complete logs at https://cloud.drone.io/leamas/shipdriver_pi/188/2/2

**Comment by mayeut** [comment]
cmake-python-distributions require cmake to build. From the logs, I'm guessing there's no previous installation of cmake.
see also #33 which might be relevant.

**Comment by leamas** [comment]
Sorry for long delay...

Nope, not that simple. cmake is present, although an old version.  Also, there is a seemingly undeclared build dependency on scikit-build package which needs to be installed manually, otherwise the build breaks. Current error message:
```
 ERROR: Command errored out with exit status 1:
205s
1900   command: /usr/bin/python3 /root/.local/lib/python3.5/site-packages/pip/_vendor/pep517/_in_process.py build_wheel /tmp/tmphpi4j6ti
205s
1901       cwd: /tmp/pip-install-ka0g08dv/cmake_36849a7ce17247d8a0ce2ae47eb8c1e5
205s
1902  Complete output (9 lines):
205s
1903    File "/tmp/pip-build-env-_s0r4lso/overlay/lib/python3.5/site-packages/skbuild/setuptools_wrap.py", line 564, in setup
205s
1904      cmkr = cmaker.CMaker(cmake_executable)
205s
1905    File "/tmp/pip-build-env-_s0r4lso/overlay/lib/python3.5/site-packages/skbuild/cmaker.py", line 95, in __init__
205s
1906      self.cmake_version = get_cmake_version(self.cmake_executable)
205s
1907    File "/tmp/pip-build-env-_s0r4lso/overlay/lib/python3.5/site-packages/skbuild/cmaker.py", line 82, in get_cmake_version
205s
1908      "Problem with the CMake installation, aborting build. CMake executable is %s" % cmake_executable)
```

Complete build logs [here](https://cloud.drone.io/leamas/shipdriver_pi/194/1/2)

**Comment by leamas** [comment]
@rhastie:

Looks like the same bug. 
For now, we work around by using pre-compiled binary packages for now, the build script has:

```
# Temporary fix until 3.19+  is available as a pypi package
# 3.19 is needed: https://gitlab.kitware.com/cmake/cmake/-/issues/20568
url='https://dl.cloudsmith.io/public/alec-leamas/opencpn-plugins-stable/deb/debian'
wget $url/pool/${OCPN_TARGET/-*/}/main/c/cm/cmake-data_3.19.3-0.1_all.deb
wget $url/pool/${OCPN_TARGET/-*/}/main/c/cm/cmake_3.19.3-0.1_armhf.deb
sudo apt install ./cmake_3.19.3-0.1_armhf.deb ./cmake-data_3.19.3-0.1_all.deb
```

$OCPN_TARGET expands to "buster"

**Comment by ericLemanissier** [comment]
Same thing happening here on raspberry os, since cmake 3.22.1 is on pypi:
https://github.com/bincrafters/system-packages-checks/runs/4496926552?check_suite_focus=true#step:3:386
```
+ python3 -m pip install cmake
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting cmake
  Downloading cmake-3.22.1.tar.gz (30 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Building wheels for collected packages: cmake
  Building wheel for cmake (pyproject.toml): started
  Building wheel for cmake (pyproject.toml): finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 /usr/local/lib/python3.7/dist-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /tmp/tmphz12lnuh
       cwd: /tmp/pip-install-bmfk_sgw/cmake_655a3adb41a942308978185a07623cb8
  Complete output (9 lines):
    File "/tmp/pip-build-env-xxr67zw1/overlay/lib/python3.7/site-packages/skbuild/setuptools_wrap.py", line 564, in setup
      cmkr = cmaker.CMaker(cmake_executable)
    File "/tmp/pip-build-env-xxr67zw1/overlay/lib/python3.7/site-packages/skbuild/cmaker.py", line 95, in __init__
      self.cmake_version = get_cmake_version(self.cmake_executable)
    File "/tmp/pip-build-env-xxr67zw1/overlay/lib/python3.7/site-packages/skbuild/cmaker.py", line 82, in get_cmake_version
      "Problem with the CMake installation, aborting build. CMake executable is %s" % cmake_executable)
  Traceback (most recent call last):
  
  Problem with the CMake installation, aborting build. CMake executable is cmake
  ----------------------------------------
  ERROR: Failed building wheel for cmake
```
The build passed 1 hour earlier, when it got cmake 3.22.0

**Comment by henryiii** [comment]
I don't see anything here: https://github.com/scikit-build/cmake-python-distributions/compare/3.22.0...3.22.1 that seems likely to break?

**Comment by henryiii** [comment]
Ahh, [piwheels](https://www.piwheels.org). I think their CI just takes a little time after a release. Tell pip to require binaries and that will fix the small window after release issue.

**Comment by ericLemanissier** [comment]
indeed: https://www.piwheels.org/project/cmake/

**Comment by henryiii** [comment]
`PIP_ONLY_BINARY=cmake` should fix the problem.

**Comment by mayeut** [comment]
With binary wheels now available for for armv7l Ubuntu 20.04+/Debian 11+ on PyPI (in addition to piwheel) and the bootstrap build added in #568, this should be solved (probably slow if building from sources on RPi).
